From bd703d405f0b214b52d63b8fed3fde5ec69f3573 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Wed, 20 Dec 2017 21:43:00 +0100
Subject: [PATCH 037/249] wiki: Link headers properly

* Instead of making the whole header-area a link (which overlaps other
  content by using margin and padding), add a font element to be used
  as anchor

Change-Id: Icaa1117852cebd00e321a78a29421964a7c6d087
---
 css/customstyles.css |  4 ++++
 js/customscripts.js  |  9 +++++++--
 js/toc.js            | 14 +++++++-------
 3 files changed, 18 insertions(+), 9 deletions(-)

diff --git a/css/customstyles.css b/css/customstyles.css
index 1483abd..fe94515 100644
--- a/css/customstyles.css
+++ b/css/customstyles.css
@@ -1232,3 +1232,7 @@ h4.panel-title {
 .navbar-fixed-bottom .navbar-collapse, .navbar-fixed-top .navbar-collapse {
     max-height: 90vH;
 }
+
+.clickable-header {
+    cursor: pointer;
+}
diff --git a/js/customscripts.js b/js/customscripts.js
index 9b67336..a5ed4b8 100644
--- a/js/customscripts.js
+++ b/js/customscripts.js
@@ -1,8 +1,13 @@
 $('#mysidebar').height($(".nav").height());
 
 $( document ).ready(function() {
-     // AnchorJS
-    anchors.add('h2,h3,h4,h5');
+    $("h2,h3,h4,h5,h6").each(function(_, headers) {
+        if (headers.firstChild.nodeType == Node.TEXT_NODE) {
+            headers.innerHTML = "<font class='clickable-header'>" + headers.innerHTML + "</font>";
+        }
+    });
+    // AnchorJS
+    anchors.add('.clickable-header');
 });
 
 // needed for nav tabs on pages. See Formatting > Nav tabs for more details.
diff --git a/js/toc.js b/js/toc.js
index a54d172..a0dd6ff 100644
--- a/js/toc.js
+++ b/js/toc.js
@@ -42,13 +42,13 @@
     var level = get_level(headers[0]),
       this_level,
       html = settings.title + " <"+settings.listType+">";
-    headers.on('click', function() {
-      if (!settings.noBackToTopLinks) {
-        window.location.hash = this.id;
-      }
-    })
-    .addClass('clickable-header')
-    .each(function(_, header) {
+    headers.each(function(_, header) {
+      var child = header.firstChild;
+      $(child).on('click', function() {
+        if (!settings.noBackToTopLinks) {
+          window.location.hash = header.id;
+        }
+      });
       this_level = get_level(header);
       if (!settings.noBackToTopLinks && this_level === highest_level) {
         $(header).addClass('top-level-header').after(return_to_top);
-- 
2.17.1

